equipment_handler/source/adc_handler.c

```diff
@@ -40,49 +41,13 @@ unsigned char adc_init(uint8_t slave_addr, uint8_t channel) {
 }
 
 void adc_write(uint8_t *buf, int size, uint8_t slave_addr) {
-    i2cSetSlaveAdd(ADC_i2c_PORT, slave_addr);
-    i2cSetDirection(i2cREG1, I2C_TRANSMITTER);
-    i2cSetBaudrate(ADC_i2c_PORT, 400);
-    i2cSetCount(ADC_i2c_PORT, size);
-    /* Set mode as Master */
-    i2cSetMode(i2cREG1, I2C_MASTER);
-
-
-    i2cSetStop(ADC_i2c_PORT);
-    /* Transmit Start Condition */
-    i2cSetStart(ADC_i2c_PORT);
-    i2cSend(ADC_i2c_PORT, size, buf);
-
-    /* Wait until Bus Busy is cleared */
-    while(i2cIsBusBusy(ADC_i2c_PORT) == true);
-    i2cSetStop(ADC_i2c_PORT);
-    /* Wait until Stop is detected */
-    //while(i2cIsStopDetected(ADC_i2c_PORT) == 0);
-    /* Clear the Stop condition */
-    i2cClearSCD(ADC_i2c_PORT);
+    // TODO: Make this use error code return
+    i2c_Send(ADC_i2c_PORT, slave_addr, size, buf);
```

Comment:
Do you mean we check if it sends successfully?

Reply:
Yeah, it would require a pretty big rewrite for the rest of the code to handle, so I'll make it an issue after this is merged